src/rpc/response.rs

```diff
@@ -54,13 +54,15 @@ impl Response {
 			span.record("rpc.error_code", err.code);
 			span.record("rpc.error_message", err.message.as_ref());
 		}
+		// Cheaper to clone the id in case of a failure
+		// than to clone the entire response, which can be arbitrary size
+		let id = self.id.clone();
```

Comment:
We want to clone the `id` only if there is an error.

I would suggest to cloning it in the Err(_) branch

Reply:
At that point `self` is already consumed, which is why I had to place it outside the match statement :)